### PR TITLE
add_dockerfile() now use installed package version and explicit System Requirements

### PR DESCRIPTION
**Pull request by VincentGuyader**
## New features

+ `add_dockerfile()` now use installed package version and explicit System Requirements

### CODE REVIEW

**Comment by VincentGuyader** [comment]
https://github.com/r-hub/sysreqs is not on CRAN...

**Comment by VincentGuyader** [comment]
we can use this : 


```
#' get system requirements
#'
#' @param packages 
#'
#' @export
# @importFrom desc desc_get_deps
#' @importFrom miniCRAN pkgDep
#' @importFrom utils download.file
#' @importFrom jsonlite fromJSON
get_sysreqs <- function(packages){
# packages <- desc::desc_get_deps(path)$package
all_deps <- paste(miniCRAN::pkgDep(packages,suggests = FALSE,quiet=FALSE), collapse = ",")
url <- sprintf("https://sysreqs.r-hub.io/pkg/%s/linux-x86_64-debian-gcc",all_deps)
path <- tempfile()
utils::download.file(url, path,mode = "wb")
out <- jsonlite::fromJSON(path)
unlink(path)
out
}

```

**Comment by VincentGuyader** [comment]
I replace `miniCRAN::pkgDep` with ` tools::package_dependencies`

**Comment by ColinFay** [comment]
Ok so the issue here is that if I have locally / or package from GitHub installed, they are put into the Dockerfile. 
I think we should do something to prevent that 🤔

For example, if I run on a local app, I have `RUN Rscript -e 'remotes::install_version("golem", version = "0.1.0.9001")'` which will fail.

**Comment by ColinFay** [comment]
We should use :

``` r 
> remotes::dev_package_deps()
Needs update -----------------------------
 package installed    available    is_cran remote
 tidyr   4618116a9... e8c3f23c4... FALSE   GitHub
Not on CRAN ----------------------------
 package installed  available is_cran remote
 golem   0.1.0.9001 NA        TRUE    CRAN  
 ggplot2 3.2.1      3.2.0     TRUE    CRAN  
 Rcpp    1.0.2      1.0.1     TRUE    CRAN  

> x <- remotes::dev_package_deps()
> tibble::as_tibble(x)
# A tibble: 63 x 6
   package   installed                           available                           diff is_cran remote   
   <chr>     <chr>                               <chr>                              <int> <lgl>   <list>   
 1 shiny     1.3.2                               1.3.2                                  0 TRUE    <cran_rm…
 2 golem     0.1.0.9001                          NA                                     2 TRUE    <cran_rm…
 3 processx  3.4.0                               3.4.0                                  0 TRUE    <cran_rm…
 4 attempt   0.3.0                               0.3.0                                  0 TRUE    <cran_rm…
 5 DT        0.7                                 0.7                                    0 TRUE    <cran_rm…
 6 glue      1.3.1                               1.3.1                                  0 TRUE    <cran_rm…
 7 htmltools 0.3.6                               0.3.6                                  0 TRUE    <cran_rm…
 8 ggplot2   3.2.1                               3.2.0                                  1 TRUE    <cran_rm…
 9 dplyr     0.8.3                               0.8.3                                  0 TRUE    <cran_rm…
10 tidyr     4618116a9296e0f11337c416966ea0f5a6… e8c3f23c46025941774c269eb547cffea…    -1 FALSE   <gthb_rm…
# … with 53 more rows
```

And build on top of that

**Comment by ColinFay** [comment]
This would also be useful to return info about locally installed packages when returning the dockerfile so that the users know they have to manually add it

``` r 
> remotes::dev_package_deps()
remotes::dev_package_deps()
Needs update -----------------------------
 package      installed    available    is_cran remote
 hiddennampkg 0.1.3        NA           FALSE   local 
 tidyr        4618116a9... e8c3f23c4... FALSE   GitHub
Not on CRAN ----------------------------
 package installed  available is_cran remote
 golem   0.1.0.9001 NA        TRUE    CRAN  
 Rcpp    1.0.2      1.0.1     TRUE    CRAN  
 openssl 1.4.1      1.4       TRUE    CRAN  
 ggplot2 3.2.1      3.2.0     TRUE    CRAN  
 sys     3.3        3.2       TRUE    CRAN  
```

**Comment by VincentGuyader** [comment]
ready to be merged...

File: DESCRIPTION [hunk 1]
```diff
@@ -42,38 +42,40 @@ Imports:
     cli,
     crayon,
     desc,
+    devtools,
```
Comment (ColinFay) [inline DESCRIPTION]: I think we don't need devtools here

**Pull request merged**